mlflow/tracking/fluent.py

```diff
@@ -702,6 +715,48 @@ def log_params(params: Dict[str, Any]) -> None:
     MlflowClient().log_batch(run_id=run_id, metrics=[], params=params_arr, tags=[])
 
 
+def log_input(dataset: Dataset, context: str) -> None:
+    """
+    Log a dataset used in the current run.
+
+    :param dataset: mlflow.data.Dataset object to be logged.
+    :param context: Context in which the dataset is used. For example: "training", "testing".
+    :returns: None
+
+    .. test-code-block:: python
+        :caption: Example
+
+        import mlflow
+
+        df = pd.read_csv("data.csv")
+        dataset = mlflow.data.from_pandas(df, source="data.csv")
+
+        # Log an input dataset used for training
+        with mlflow.start_run():
+            mlflow.log_input(dataset, context="training")
+    """
+    run_id = _get_or_start_run().info.run_id
+    dataset_dict = json.loads(dataset.to_json())
+    dataset_input = DatasetInput(
+        dataset=DatasetEntity(
+            name=dataset_dict["name"],
+            digest=dataset_dict["digest"],
+            source_type=dataset_dict["source_type"],
+            source=dataset_dict["source"],
+            schema=dataset_dict.schema if "schema" in dataset_dict else None,
+            profile=dataset_dict.profile if "profile" in dataset_dict else None,
+        ),
```

Comment:
Can we define a `_to_mlflow_entity()` method on the dataset base class that we can call here?